### PR TITLE
manticoresearch: 5.0.3 -> 6.0.4

### PR DESCRIPTION
**Pull request by DChalcraft**
###### Description of changes

Update manticore from 5.0.3 to 6.0.4 - [changelog](https://manual.manticoresearch.com/Changelog#Version-6.0.4).

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by jdelStrother — CHANGES_REQUESTED** [review]
@DChalcraft because the sha256 values for columnar & manticoresearch are unchanged, this just fetches the old versions out of the nix store and rebuilds them - if you run `result/bin/searchd --version` you'll see 5.0.3 in the output.

It's possible to precompute what the sha256 will be with `nix-prefetch-url`, but honestly I usually just tweak a single character of the existing hash, try & build, and then re-update the hash from the resulting error message.

**Comment by DChalcraft** [comment]
>because the sha256 values for columnar & manticoresearch are unchanged, this just fetches the old versions out of the nix store and rebuilds them

Ahh, thanks - I assumed there was something wrong with my setup so spent some time removing and rebuilding nixpkgs etc as the output of `nix-shell -p nixpkgs-review  --run "nixpkgs-review pr https://github.com/NixOS/nixpkgs/pull/237929"` seemed to imply that manticoresearch had been rebuilt, but the version was still 5.0.3.

**Comment by jdelStrother** [comment]
This fails under sandboxing because it tries to download http://github.com/nlohmann/json/archive/refs/tags/v3.10.5.zip

Depending on how manticore have set up their make files, you might be able to fix it just by adding `nlohmann_json` into the buildInputs.

File: pkgs/servers/matrix-synapse/tools/synadm.nix [hunk 1]
```diff
@@ -6,12 +6,12 @@
...
 python3.pkgs.buildPythonApplication rec {
   pname = "synadm";
-  version = "0.41.2";
+  version = "0.41.3";
   format = "setuptools";
 
   src = fetchPypi {
     inherit pname version;
-    hash = "sha256-wSpgc1umBMLCc2ThfYSuNNnzqWXyEQM0XhTuOAQaiXg=";
+    hash = "sha256-gWEgLpSE77XdocAZqN1i/vR5dvYFsgsg5zs5Dj90V/o=";
```
Comment (jdelStrother) [inline pkgs/servers/matrix-synapse/tools/synadm.nix:14]: This commit is presumably an accidental inclusion? Could you rebase to remove it?

**Review by jdelStrother — CHANGES_REQUESTED** [review]
LGTM, other than the accidental synadm commit

**Review by jdelStrother — APPROVED** [review]
lgtm

**Comment by DChalcraft** [comment]
Can anyone merge this PR?

**Comment by onny** [comment]
Build currently fails for me :(

```
manticoresearch> -- Build files have been written to: /build/source/build
manticoresearch> cmake: enabled parallel building
manticoresearch> cmake: enabled parallel installing
manticoresearch> building
manticoresearch> build flags: -j8 SHELL=/nix/store/xfb3ykw9r5hpayd05sr0cizwadzq1d8q-bash-5.2-p15/bin/bash
manticoresearch> installing
manticoresearch> install flags: -j8 SHELL=/nix/store/xfb3ykw9r5hpayd05sr0cizwadzq1d8q-bash-5.2-p15/bin/bash install
manticoresearch> make: *** No rule to make target 'install'.  Stop.
```

**Comment by jdelStrother** [comment]
@onny odd, it seemed ok for me on aarch64-darwin & aarch64-linux (under vmware). What're you on?
Here's a build log: https://gist.github.com/jdelStrother/2e48407d5da41c22ef682a1aecb1cb59

**Comment by onny** [comment]
I tried to build it on linux-x86_64, but maybe I was missing something. I'll look into it

**Comment by drupol** [comment]
Hi !

Thanks for your contribution.

Keep them coming :)

**Comment by onny** [comment]
Result of `nixpkgs-review pr 237929` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>manticoresearch</li>
  </ul>
</details>

**Comment by onny** [comment]
``` 
error: builder for '/nix/store/apgfjw02i85gb2zvdrn2ib2z1sdihfi2-manticoresearch-6.0.4.drv' failed with exit code 2;
       last 10 log lines:
       > -- Installing: /nix/store/wijf6aj6mnfsgqnjdvisxvvqa47px7rq-manticoresearch-6.0.4/bin/searchd
       > CMake Error at src/cmake_install.cmake:95 (file):
       >   file INSTALL cannot copy file "/build/source/build/src/searchd" to
       >   "/nix/store/wijf6aj6mnfsgqnjdvisxvvqa47px7rq-manticoresearch-6.0.4/bin/searchd":
       >   No space left on device.
       > Call Stack (most recent call first):
       >   cmake_install.cmake:67 (include)
       >
       > 
       > make: *** [Makefile:120: install] Error 1
       For full logs, run 'nix log /nix/store/apgfjw02i85gb2zvdrn2ib2z1sdihfi2-manticoresearch-6.0.4.drv'.
       note: build failure may have been caused by lack of free disk space
error: 1 dependencies of derivation '/nix/store/icq1f1wf09sd5qfyi9rv4j4fqxbdadis-review-shell.drv' failed to build
```
strange that this only happens to me :(

**Comment by onny** [comment]
Result of `nixpkgs-review pr 237929` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>manticoresearch</li>
  </ul>
</details>

**Comment by onny** [comment]
Result of `nixpkgs-review pr 237929` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>manticoresearch</li>
  </ul>
</details>

**Comment by onny** [comment]
Thanks for fixing, reviewing and merging :)

**Comment by jdelStrother** [comment]
@onny did @drupol ’s changes make it build for you? I’m not sure I get why, was it fixed by something in master?

**Comment by drupol** [comment]
No the changes I did has no impact at all on the failures if any. It's just a better way to write derivations, nothing else.

**Comment by onny** [comment]
> @onny did @drupol ’s changes make it build for you? I’m not sure I get why, was it fixed by something in master?

Could be an issue on my side, sorry. Probably full disk space

**Pull request merged**